### PR TITLE
Added new CPU Manager topic

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://trello.com/c/Bw2KTNJl/637-5-psap-cpu-manager-documentation

Preview build: http://file.rdu.redhat.com/~ahardin/12202017/cpu-manager/scaling_performance/using_cpu_manager.html

@jeremyeder PTAL

### CODE REVIEW

File: scaling_performance/using_cpu_manager.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[what-cpu-manager-does]]
+== What CPU Manager Does
+
+CPU Manager manages groups of CPUs and constrains workloads to specific CPUs.
```
Comment (ahardin-rh) [inline scaling_performance/using_cpu_manager.adoc]: @jeremyeder What are some specific use cases for CPU manager? When is it used and why do users care?

Comment (jeremyeder) [reply scaling_performance/using_cpu_manager.adoc]: This feature is useful for workloads that have some of these attributes:

* Require as much CPU time as possible (eliminate jitter).
* Are sensitive to processor cache misses.
* Are low-latency network applications.
* Coordinate with other processes and benefit from sharing a single processor cache.

**Comment by jeremyeder** [comment]
The comments from redhat-performance-bot are me.  Was logged into the wrong github account.

**Comment by ahardin-rh** [comment]
@jeremyeder I always knew you were a robot. 🤖 Thanks for your comments!

**Comment by jeremyeder** [comment]
Is there a new copy of post-processed docs I could see?

**Comment by ahardin-rh** [comment]
@jeremyeder This is now updated: http://file.rdu.redhat.com/~ahardin/12202017/cpu-manager/scaling_performance/using_cpu_manager.html

Thanks!

File: scaling_performance/using_cpu_manager.adoc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
++
+You can see that `Allocatable CPU` is 3500 millicores. This means we can run three of
+our CPU manager pods since each will take one whole core. A whole core is
+equivalent to 1000 millicores.
++
+If you try to schedule a fourth pod, the system will accept the pod, but it will
+never be scheduled:
++
+----
+# oc get pods --all-namespaces |grep grafana
```
Comment (jeremyeder) [inline scaling_performance/using_cpu_manager.adoc]: s/grafana/test

Comment (ahardin-rh) [reply scaling_performance/using_cpu_manager.adoc]: @jeremyeder Fixed. Thanks! Is this okay for 3.7 (next weekly release), or should we hold this for 3.9?

Comment (jeremyeder) [reply scaling_performance/using_cpu_manager.adoc]: 3.9 only

**Comment by derekwaynecarr** [comment]
LGTM

File: scaling_performance/using_cpu_manager.adoc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  test                        cpumanager-r9wrq          1 (28%)       1 (28%)     1G (6%)          1G (6%)
+...
+Allocated resources:
+  (Total limits may be over 100 percent, i.e., overcommitted.)
+  CPU Requests  CPU Limits  Memory Requests  Memory Limits
+  ------------  ----------  ---------------  -------------
+  3 (85%)       3 (85%)     5437500k (32%)   9250M (55%)
+----
++
+This VM has four CPU cores. You set `kube-reserved` to 500 millicores, meaning
```
Comment (ravisantoshgudimetla) [inline scaling_performance/using_cpu_manager.adoc]: Didn't we set this value to 1000 earlier in https://github.com/openshift/openshift-docs/pull/6750/files#diff-ac5bd8640a40585a2f744a1fcdc79fb4R57

File: scaling_performance/using_cpu_manager.adoc [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+kubeletArguments:
+...
+  feature-gates:
+  - CPUManager=true
+  cpu-manager-policy:
+  - static
+  cpu-manager-reconcile-period:
+  - 5s
+  kube-reserved: <1>
+  - cpu=1000m
```
Comment (jeremyeder) [inline scaling_performance/using_cpu_manager.adoc]: @ahardin-rh please change 1000m to 500m

File: scaling_performance/using_cpu_manager.adoc (hunk 3 above)
Comment (jeremyeder) [reply scaling_performance/using_cpu_manager.adoc]: Good catch.  Must be a copy paste error on my part.  Let's just change that to 500m.  I put a comment to ask @ahardin-rh to do that.

**Comment by ahardin-rh** [comment]
@jeremyeder Thanks! Updated:
http://file.rdu.redhat.com/%7Eahardin/12202017/cpu-manager/scaling_performance/using_cpu_manager.html

**Review by ravisantoshgudimetla — APPROVED** [review]
LGTM

**Comment by ahardin-rh** [comment]
This is now QE verified.
@openshift/team-documentation Please peer review.

File: scaling_performance/using_cpu_manager.adoc [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+CPU Manager is useful for workloads that have some of these attributes:
+
+* Require as much CPU time as possible.
+* Are sensitive to processor cache misses.
+* Are low-latency network applications.
+* Coordinate with other processes and benefit from sharing a single processor
+cache.
+
+[[setting-up-cpu-manager]]
+== Setting Up CPU Manager
```
Comment (nengard) [inline scaling_performance/using_cpu_manager.adoc]: Up should be lowercase.

File: scaling_performance/using_cpu_manager.adoc [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+QoS tiers end up in child `cgroups` of `kubepods`.
++
+----
+# cd /sys/fs/cgroup/cpuset/kubepods.slice/kubepods-pod0ec1ab8b_e1c4_11e7_bb22_027b30990a24.slice/docker-b24e29bc4021064057f941dc5f3538595c317d294f2c8e448b5e61a29c026d1c.scope
+# for i in `ls cpuset.cpus tasks` ; do echo -n "$i "; cat $i ; done
+cpuset.cpus 2
+tasks 44216
+----
+
+. Check the task's allowed CPU list:
```
Comment (nengard) [inline scaling_performance/using_cpu_manager.adoc]: IBM Styleguide suggests avoiding possessives.  Maybe " Check the allows CPU list for the task:"

File: scaling_performance/using_cpu_manager.adoc [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
++
+----
+# grep ^Cpus_allowed_list /proc/44216/status
+Cpus_allowed_list:      2
+----
+
+. Verify that another pod (in this case, the pod in the `burstable` QoS tier) on the
+system cannot run on the core allocated for the `Guaranteed` pod:
```
Comment (nengard) [inline scaling_performance/using_cpu_manager.adoc]: can not - two words per IBM Stylguide

File: scaling_performance/using_cpu_manager.adoc [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+...
+Allocated resources:
+  (Total limits may be over 100 percent, i.e., overcommitted.)
+  CPU Requests  CPU Limits  Memory Requests  Memory Limits
+  ------------  ----------  ---------------  -------------
+  3 (85%)       3 (85%)     5437500k (32%)   9250M (55%)
+----
++
+This VM has four CPU cores. You set `kube-reserved` to 500 millicores, meaning
+half of one core is subtracted from the node's total capacity to arrive at the
```
Comment (nengard) [inline scaling_performance/using_cpu_manager.adoc]: 'total capacity of the node' instead of "node's"

**Review by ncbaratta — APPROVED** [review]
A few nits.

**Comment by ahardin-rh** [comment]
Fixed. Thanks @nengard

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../scaling_performance/using_cpu_manager.adoc#scaling-performance-using-cpu-manager[Using CPU Manager]
|Added a new topic on using CPU Manager to manage groups of CPUs and constrain workloads to specific CPUs.
%

**Pull request merged**